packages/file/lib/src/backends/memory/memory_file_system.dart

```diff
@@ -74,16 +74,15 @@ class _MemoryFileSystem extends FileSystem
     implements MemoryFileSystem, NodeBasedFileSystem {
   _MemoryFileSystem({
     this.style = FileSystemStyle.posix,
-    @required this.clock,
-  })  : assert(style != null),
-        assert(clock != null) {
-    _root = RootNode(this);
+    required this.clock,
+  }) {
     _context = style.contextFor(style.root);
+    _root = RootNode(this);
   }
 
-  RootNode _root;
-  String _systemTemp;
-  p.Context _context;
+  RootNode? _root;
```

Comment:
Can this be `late` non-null?

Reply:
tests say no :) .

It looks like we check whether this is null in the constructor the RootNode that we assign to it, which causes a late initialization error.